File: src/main/java/com/classy/instagram/article/entity/ArticleLike.java
package com.classy.instagram.article.entity;

import com.classy.instagram.user.entity.User;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "article_likes")
public class ArticleLike {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    private Article article;
    @ManyToOne
    private User user;
}

File: src/main/kotlin/com/classy/placelog/article/entity/ArticleLike.kt
package com.classy.placelog.article.entity

import com.classy.placelog.user.entity.User
import jakarta.persistence.Embeddable
import jakarta.persistence.EmbeddedId
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.ManyToOne

@Entity
class ArticleLike(
    @EmbeddedId
    val id: ArticleLikeId,
) {
    @Embeddable
    data class ArticleLikeId(
        @ManyToOne(fetch = FetchType.LAZY)
        val article: Article,
        @ManyToOne(fetch = FetchType.LAZY)
        val author: User,
    )

}

